libs/designer/src/lib/core/utils/outputs.ts

```diff
@@ -342,17 +349,26 @@ export const getUpdatedManifestForSchemaDependency = (manifest: OperationManifes
 
       const currentSchemaValue = getObjectPropertyValue(updatedManifest.properties.outputs, outputLocation);
 
+      const inputSchema = updatedManifest.properties?.inputs?.properties?.schema;
+      const isRequestApiConnectionTrigger =
+        'x-ms-editor-options' in inputSchema ? inputSchema['x-ms-editor-options'].isRequestApiConnectionTrigger : undefined;
+
+      let schemaValue: Schema;
+      let shouldMerge: boolean;
       // if schema contains static object returned from RP, merge the current schema value and new schema value
-      if (schemaToReplace && 'rows' in schemaToReplace) {
-        if (!Object.keys(schemaToReplace.rows.items.properties) !== undefined && 'rows' in currentSchemaValue) {
-          safeSetObjectPropertyValue(
-            updatedManifest.properties.outputs,
-            outputLocation,
-            { ...currentSchemaValue, ...schemaToReplace },
-            true
-          );
+      if (isRequestApiConnectionTrigger && schemaToReplace && 'rows' in schemaToReplace) {
+        if ('rows' in currentSchemaValue) {
+          schemaValue = { ...currentSchemaValue, ...schemaToReplace };
+          shouldMerge = true;
+        } else {
+          schemaValue = currentSchemaValue;
```

Comment:
isn't this the same as the current value. what's the point of calling `safeSetObjectPropertyValue` after this

isn't this like doing
```
const b = a;
a = b;
```

Reply:
Yeah, the else can be skipped which is why previously I didn't have an else statement. Added it incase it isn't very clear

Reply:
If this was intentional why not just do a `continue`

Reply:
`continue` makes more sense. Changed